decision-records/2024-06-04--server-to-keep-using-plain-http-while-waiting-for-cert.md

```diff
@@ -0,0 +1,61 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
+
+As a result we had some code ordering issues (e.g. #414):
+- Should we update the config file with the tls config, which includes path to
+  the cert, before we got the cert, only to perfectly reflect the juju model?
+- Should we wait with the tls config and allow the workload to continue to
+  operate in plain http, in contradiction to the juju model?
+- How should we determine within the charm if "TLS is enabled"?
+
+This problem is common to all COS charms.
+
+## Decision
+A server shall keep using plain http while waiting for cert. The charm shall
+have a "Waiting" status during that period (subject to other status
+priorities).
```

Comment:
I'm lacking context, but I vote instead for "Stop operating in plain http mode as soon as a tls-certificates relation is formed".  

Forming the `tls-certificates` relation is an indication by an admin that they want TLS.  As an admin if I turn TLS "on" I expect only secure traffic to be allowed now, not a best-effort "we'll try to secure traffic".  The failure mode if we continue to operate in plain http is too risky too - in the case of anything going wrong (we never receive a cert, some other funny thing) I'd want my application to break by rejecting everything rather than by allowing unsecured traffic

Reply:
Done.